### PR TITLE
DCMAW-7747: Remove github.properties; Use gradle home props

### PR DESCRIPTION
**Pull request by Chuddington**
# _DCMAW-7747: Remove github.properties; Use gradle home props_

- Use gradle properties as the primary mechanism for obtaining github credentials. The gpr.[user, token] properties should most likely be stored within the developer's `~/.gradle/gradle.properties` file.

  - gpr.user is your Github account name.
  - gpr.token is your personal access token, generated at https://github.com/settings/tokens. The proceeding terminal command also works: `gh auth token`

- Rename system environment variable names for github packages registry credentials. Workflow input variables have also been renamed:

  - USERNAME environment variable to GITHUB_ACTOR
  - TOKEN environment variable to GITHUB_TOKEN
  - access-token github workflow input to github-token
  - username github workflow input to github-actor

- Create github-authentication.md for documenting the necessary steps.

### CODE REVIEW

File: .github/workflows/on_pull_request.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ jobs:
   onPullRequestJob:
     name: Verify code base when pull request is published/updated
     runs-on: macos-latest
+    permissions:
+      # Set permissions for ${{ secrets.GITHUB_TOKEN }}
+      # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+      packages: read
```
Comment (andyloughran) [inline .github/workflows/on_pull_request.yml:28]:
```suggestion
      contents: read
      packages: read
```

File: .github/workflows/on_push-release.yml [hunk 2]
```diff
@@ -16,6 +16,9 @@ jobs:
     runs-on: macos-latest
     #    runs-on: ubuntu-20.04-16core # Larger github runner, with KVM acceleration
     permissions:
+      # Set permissions for ${{ secrets.GITHUB_TOKEN }}
+      # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+      packages: read
```
Comment (andyloughran) [inline .github/workflows/on_push-release.yml:21]:
```suggestion
      contents: read
      packages: read
```

File: config/actions/gradle-assemble-and-bundle/action.yml [hunk 3]
```diff
@@ -2,15 +2,15 @@ name: 'Run gradle assemble and bundle (release build type)'
 description: 'Run the "assembleRelease" and "bundleRelease" commands for a Gradle project'
 
 inputs:
-  access-token:
+  github-token:
```
Comment (andyloughran) [inline config/actions/gradle-assemble-and-bundle/action.yml:5]: We can get rid of these right - they'll be set at the workflow level so don't need to ingest them as inputs.

Comment (andyloughran) [reply config/actions/gradle-assemble-and-bundle/action.yml:5]: Ignore me, these apply to a custom action not a workflow_dispatch

File: .github/workflows/on_push-release.yml (hunk 2 above)
Comment (Chuddington) [reply .github/workflows/on_push-release.yml:21]: This has now been done; set to `contents: read`

File: .github/workflows/on_pull_request.yml (hunk 1 above)
Comment (Chuddington) [reply .github/workflows/on_pull_request.yml:29]: This has now been done; set to `contents: read`

**Review by alex-bradbury — APPROVED** [review]

File: docs/developer-setup/github-authentication.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+The proceeding command outputs the existing Personal Access Token, if the
+[Github Command-Line Interface (CLI) tool] is installed:
+
+```shell
+gh auth token
+```
+
+## The where: `gradle.properties`
+
+Because we no longer store credentials within the same directory structure as the git repository
```
Comment (DougsCER) [inline docs/developer-setup/github-authentication.md:31]: Small thing but best not to refer back to an old situation as that will quickly get out of date as new developers never will store credentials within the same directory. 
A more future-proof wording might be:

It is bad practice to store credentials within the same directory structure as the git repository. Instead these gradle properties should be stored within Gradle's user configuration, found on your local laptop
at `~/.gradle/gradle.properties`.

File: docs/developer-setup/github-authentication.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+itself, it's recommended to store these gradle properties within Gradle's user configuration, found
+at `~/.gradle/gradle.properties`. When successfully edited, it should look similar to the proceeding
+code block with unique values to the developer:
+
+```properties
+# Other properties within the file...
+
+gpr.user=Octocat
+gpr.token=ghp_1234567890abcdef
+```
```
Comment (DougsCER) [inline docs/developer-setup/github-authentication.md:41]: great to give an example here. thank you

File: docs/developer-setup/github-authentication.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+their set up?
+
+## The what: property keys
+
+There are two gradle properties used to authenticate with the Github packages registry for local
+development:
+
+- `gpr.user`: The developer's Github account name. Associated with the `GITHUB_ACTOR` environment
+  variable.
+- `gpr.token`: The developer's Personal Access Token (PAT), generated in the
```
Comment (DougsCER) [inline docs/developer-setup/github-authentication.md:17]: This is only relevant in private repositories right?
So we could have a link to the docs Sandy wrote about PAT token generation / deletion. 
Particularly https://govukverify.atlassian.net/wiki/x/SIIw5g

File: docs/developer-setup/github-authentication.md (hunk 4 above)
Comment (Chuddington) [reply docs/developer-setup/github-authentication.md:31]: That's understandable. However, I take issue with some of the wording you've suggested due to it not conforming with the google developer documentation style guide, such as the use of passive voice (should be stored).

I'm sure that when developers come to implement this within the various repositories, they'll find alternative ways of framing it in a way that meets everyone's expectations

File: docs/developer-setup/github-authentication.md (hunk 6 above)
Comment (Chuddington) [reply docs/developer-setup/github-authentication.md:17]: At the time of making this PR, I didn't have access to the URL you've provided. It's certainly worth adding it to the developer documentation!

Comment (Chuddington) [reply docs/developer-setup/github-authentication.md:17]: The token is seemingly required regardless of whether the repository is private

**Pull request merged**